src/smc-hub/kucalc/quota.ts

```diff
@@ -211,9 +303,27 @@ exports.quota = function(
         contribs += factor * parse_num(num);
       }
     }
+    // if we have some overcommit ratio set, increase a request
+    if (site_settings?.default_quotas != null) {
+      const { mem_oc, cpu_oc } = site_settings?.default_quotas;
+      if (
+        typeof cpu_oc == "number" &&
+        name == "cpu_request" &&
+        quota.cpu_limit != null
+      ) {
+        const oc_cpu = quota.cpu_limit / cpu_oc;
+        contribs = Math.max(contribs, oc_cpu - base);
+      } else if (
+        typeof mem_oc == "number" &&
```

Comment:
What are you REALLY testing when you write `typeof mem_oc == "number"`?  Are you really testing that at is defined at all?   So if mem_oc is accidentally a string (say), the it just gets silently ignored (not even a warning message)?  

I would probably replace `typeof mem_oc == "number"` with a test for weather `mem_oc` is true-ish, since that implies it is defined **and** nonzero.  If it were zero then you have a divide by 0 in line 321, which makes me nervous.

Reply:
I replaced this with a little wrapper to sanitize it. The point with being explicit about "number" is mainly typescript. It wants to make sure both sides of these arithmetic ops are numbers.